tests/integration/ceph_csi_test.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	csiSecretName        = "ceph-csi-secret"
+	csiSCRBD             = "ceph-csi-rbd"
+	csiSCCephFS          = "ceph-csi-cephfs"
+	csiPoolRBD           = "csi-rbd"
+	csiPoolCephFS        = "csi-cephfs"
+	csiTestRBDPodName    = "csi-test-rbd"
+	csiTestCephFSPodName = "csi-test-cephfs"
+)
+
+// a minimal mon status struct that contains monmap
+// see Ceph doc at http://docs.ceph.com/docs/mimic/rados/operations/monitoring/
+type cephMonStatus struct {
+	Name   string `json:"name"`
+	Monmap struct {
+		Mons []struct {
+			Rank       int    `json:"rank,omitempty"`
+			Name       string `json:"name,omitempty"`
+			Addr       string `json:"addr,omitempty"`
+			PublicAddr string `json:"public_addr"`
+		} `json:"mons"`
+	} `json:"monmap"`
+}
+
+func runCephCSIE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite.Suite, namespace string) {
+	if isCSIRunnig(k8sh, namespace) {
+		logger.Info("test Ceph CSI driver")
+		createCephCSISecret(helper, k8sh, s, namespace)
+		createCephPools(helper, s, namespace)
+		createCSIStorageClass(k8sh, s, namespace)
+		createCSIRBDTestPod(k8sh, s, namespace)
+	}
+}
+
+func isCSIRunnig(k8sh *utils.K8sHelper, namespace string) bool {
+	return k8sh.IsPodWithLabelPresent("app=csi-rbdplugin", installer.SystemNamespace(namespace))
```

Comment:
What's preventing the csi plugin from starting? It's a runtime check in the operator code to know whether to start the plugin pod, right?

Reply:
if csi is not enabled or kubernetes version is below 1.13, the csi drivers won't start.